src/libs/folder.js

```diff
@@ -0,0 +1,205 @@
+ * @return {Object}         Directory attributes
+ */
+export const createDirectoryByPath = async (client, path) => {
+  const { data } = await client
+    .collection(FILES_DOCTYPE)
+    .createDirectoryByPath(path)
+  return data
+}
+
+/**
+ * Retrieves a directory from its path
+ * @param  {Object}  client CozyClient
+ * @param  {string}  path   Directory path
+ * @return {Object}        Created io.cozy.files document
+ */
+export const statDirectoryByPath = async (client, path) => {
+  try {
+    const response = await client.collection(FILES_DOCTYPE).statByPath(path)
+    return response.data
+  } catch (error) {
+    if (error && error.status === 404) return null
+    throw new Error(error.message)
+  }
+}
+
+/**
+ * Build folder path for a given konnector and a given account.
+ *
+ * If konnector.folders[0].defaultDir exists, it is used as default directory.
+ *
+ * Occurrences of following strings in base directory are replaced by:
+ * * `$administrative`: Administrative folder
+ * * `$photos`: Photos folder
+ *
+ * Occurrences of following strings in path are replaced by:
+ * * `$account: Account label (id or name)`
+ * * `$konnector`: Konnector name
+ *
+ * If no konnectors.folders[0].defaultDir is set, the default dir used is
+ * *  `$administrative/$konnector/$account`
+ *
+ * @param  {Object} konnector Konnector document
+ * @param  {Object} account   Account document
+ * @param  {Object} folders   Object containing a mapping from folder
+ * identifiers (ex: $administrative) to their localized values (ex:
+ * Administratif).
+ * @return {String}           The result path
+ */
+
+export const buildFolderPath = (konnector, account, folders = {}) => {
+  const fullPath = get(
+    konnector,
+    // For now konnectors are only defining one folder in their folders array
+    'folders[0].defaultDir',
+    '$administrative/$konnector/$account'
+  )
+  // Trim `/` and avoid multiple `/` characters with regexp
+  let sanitizedPath = trim(fullPath.replace(/(\/+)/g, '/'), '/')
+  // If the konnector doesn't have any of our base dir, we set it to $administrative
+  if (!hasBaseDir(sanitizedPath)) {
+    sanitizedPath = '$administrative/' + sanitizedPath
+  }
+  /**
+   * Now that we have our sanitizedPath, we can split it in two strings
+   * * `baseDir` containing the baseDir path
+   * * `buildedSubDir` containing the rest of the path (ie the path without baseDir)
+   */
+  const baseDir = sanitizedPath.split('/', 1)
+  const buildedSubDir = buildSubDir(sanitizedPath, baseDir[0])
+
+  const renderedBaseDir = renderBaseDir(baseDir[0], folders)
+  const renderedPath = renderSubDir(buildedSubDir, {
+    // When adding a new allowed variable here, please keep documentation
+    // of `renderSubDir` function up to date.
+    konnector: konnector.name,
+    account: models.accounts
+      .getAccountName(account)
+      .replace(sanitizeAccountIdentifierRx, '-')
+  })
+  return `/${renderedBaseDir}/${renderedPath}`
+}
```

Comment:
cozy-client konnector model please. This is a copy paste from harvest we should mutualize.

Reply:
If this is what we mentioned during our daily, then this is planned. Here the goal is to unlock other team tasks faster, and then the code should be mutualised somewhere like cozy-client after this PR is merged.

However during our daily the CI was still not working. Now that it is fixed I don't know if this impacts our previous decision.